### PR TITLE
First version

### PR DESCRIPTION
**Pull request by boskokul**

### CODE REVIEW

File: BufferGame/Buffer.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+        public Buffer(Logger logger)
+        {
+            Values = new List<int>();
+            ValuesNew = new List<CollectionDescription>();
+            Logger = logger;
+        }
+
+
+        public void AddDataNew(Code code, int dataValue)
```
Comment (odbeast) [inline BufferGame/Buffer.cs:20]: Ovde radi citljivosti i odrzavanja code-a bi isao nesto ovako 

 public void AddDataNew(Code code, int dataValue)
        {
              if (CodeExists(code))
              {
                  UpdateExistingBufferProperty(code, dataValue);
              }
              else if (DatasetExists(code))
              {
                  CreateNewBufferPropertyInDataset(code, dataValue);
              }
              else
              {
                  CreateNewCollectionDescription(code, dataValue);
              }
        }
        
        gde ovi foreach-evi idu u ove manje metode (ne mora takav name-ing)

File: BufferGame/Buffer.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                            SendToHistoricalData(cd);
+                            cd.BufferPropertyCollection.Clear();
+                        }
+                        return;
+                    }
+                }
+            }
+
+            var collectionDescription = new CollectionDescription();
+            collectionDescription.Dataset = DatasetCodes[code];
```
Comment (odbeast) [inline BufferGame/Buffer.cs:56]: DatasetCodes[code];

Stoji da ce u ovom slucaju uvek biti popunjem Dictionary, al neka praksa kod nas je da se uvek radi TryGetProperty. Al popricamo o tome u petak jos.

File: BufferGame/HistoricalData.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            Logger = logger;
+            Values = new List<int>();
+            ListDescription = new List<Description>();
+        }
+
+
+        public void StoreDataFromBufferNew(CollectionDescription collectionDescription)
+        {
+            // provera da li je odgovarajuc dataset sa kodovima
+            if (ValidateData(collectionDescription) == false)
```
Comment (odbeast) [inline BufferGame/HistoricalData.cs:23]: if (!ValidateData(collectionDescription))

File: BufferGame/HistoricalData.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            {
+                Logger.Log($"Dataset and code not valid\n");
+                return;
+            }
+            foreach (var description in ListDescription)
```
Comment (odbeast) [inline BufferGame/HistoricalData.cs:28]: foreach (var description in ListDescription)
        {
            if (description.Dataset == collectionDescription.Dataset)
            {
                ProcessExistingDataset(description, collectionDescription.BufferPropertyCollection);
                return;
            }
        }

        ProcessNewDataset(collectionDescription);
    }
    
    isto razdvojiti u manje metode, lakse je posle snaci se, a ako stavimo i logove u te iste metode lakse mozemo pronaci gde je bug

File: BufferGame/HistoricalData.cs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                if (GlobalData.DatasetCodes[bf.Code] != collectionDescription.Dataset)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+
+        public void StoreDataFromWritterNew(Code code, int dataValue)
```
Comment (odbeast) [inline BufferGame/HistoricalData.cs:87]: Isto razdvojiti 
 public void StoreDataFromWritterNew(Code code, int dataValue)
    {
        if (DatasetExists(code))
        {
            ProcessExistingDatasetForCode(code, dataValue);
        }
        else
        {
            ProcessNewDatasetForCode(code, dataValue);
        }
    }

File: BufferGame/HistoricalData.cs [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+            desc.ID = Guid.NewGuid();
+            desc.Dataset = DatasetCodes[code];
+            Logger.Log($"Storing historical data -> \n ID \t\t\t\t Dataset |  code \t|  value \t \n {desc.ID} {desc.Dataset} | {code} | {dataValue}");
+            
+            // ovde cuvati u bazu
+            desc.HistoricalProperties.Add(new HistoricalProperty(code, dataValue));
+            ListDescription.Add(desc);
+        }
+
+        public bool DatasetExists(Code code)
```
Comment (odbeast) [inline BufferGame/HistoricalData.cs:123]: public bool DatasetExists(Code code)
    {
        return ListDescription.Any(description => description.Dataset == DatasetCodes[code]);
    }

File: BufferGame/Program.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+            int userInput = 0;
+            string input = Console.ReadLine();
+            while (!string.IsNullOrWhiteSpace(input))
+            {
+                if (input.Equals("s"))
+                {
+                    writter.SendDataToHistoricalDataNew();
+                }
+                if (int.TryParse(input, out userInput))
```
Comment (odbeast) [inline BufferGame/Program.cs:29]: mozda else if al videcemo u petak

Comment (odbeast) [reply BufferGame/Program.cs:29]: string input;
do
{
    input = Console.ReadLine();
    if (input.Equals("s"))
    {
        writter.SendDataToHistoricalDataNew();
    }
    else if (int.TryParse(input, out int userInput) && userInput >= 1 && userInput <= 8)
    {
        reader.RequestData(userInput);
    }
} while (!string.IsNullOrWhiteSpace(input));

File: BufferGame/Buffer.cs (hunk 2 above)
Comment (boskokul) [reply BufferGame/Buffer.cs:56]: Vazi, hvala.

File: BufferGame/HistoricalData.cs (hunk 6 above)
Comment (boskokul) [reply BufferGame/HistoricalData.cs:123]: Hvala! 👍

**Pull request merged**